sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -427,6 +430,44 @@ private[hive] class HiveClientImpl(
     client.alterTable(qualifiedTableName, hiveTable)
   }
 
+  override def alterTableStats(table: CatalogTable): Unit = withHiveState {
```

Comment:
Currently the `HiveClient` is responsible to talk to hive metastore and convert hive meta(HiveTable) to Spark meta(CatalogTable).

The `HiveExternalCatalog` is responsible to act as bridge between external catalog interface and hive client, and do some tricks if the hive client can't support the functionality defined by external catalog.